### PR TITLE
Route name, controller/action, controller based configuration

### PR DESCRIPTION
**Pull request by corentin-larose**
Allowing to use zf-http-cache with ZF2 by making configuration either route or controller/action or only controller based.

### CODE REVIEW

File: src/HttpCacheListener.php [hunk 1]
```diff
@@ -127,16 +127,34 @@ public function onRoute(MvcEvent $e)
...
+        } elseif (! empty($cacheConfig[$controller])) {
             $controllerConfig = $cacheConfig[$controller];
         } elseif (! empty($this->config['regex_delimiter'])) {
             foreach ($cacheConfig as $key => $config) {
                 if (substr($key, 0, 1) === $this->config['regex_delimiter']) {
-                    if (preg_match($key, preg_quote($controller, $this->config['regex_delimiter']))) {
+                    if (
+                        preg_match($key, $routeName)
+                        or preg_match($key, "$controller::$action")
+                        or preg_match($key, $controller)
```
Comment (weierophinney) [inline src/HttpCacheListener.php]: Use `||` instead of `or`, please.

Also, the first condition should immediately follow the opening brace.

**Comment by weierophinney** [comment]
This looks like it may be complete, judging based on the tests; all it needs is a little additional documentation.

I can do that on merge, and fix the CS issues, if you like @corentin-larose — do you feel it's ready?

**Comment by corentin-larose** [comment]
Hmm, I'm pretty sure that I wrote that documentation already, but I had to switch on another task before I finished... Gonna try to find what it became !

**Comment by corentin-larose** [comment]
@weierophinney I just added some doc, feel free to improve if necessary.

Since zf-http-cache is now compatible with a classical MVC approach, we could add it in the suggested packages in Zend Framework, don't you think?

**Pull request merged**